python-lib/dku_deeplearning_image/config_handler.py

```diff
@@ -22,7 +22,8 @@ def add_gpu_config(dku_config, config):
         }])
     dku_config.add_param(
         name='gpu_memory_allocation_mode',
-        value=config.get('gpu_memory_allocation_mode'))
+        value=config.get('gpu_memory_allocation_mode'),
+        cast_to=constants.GPU_MEMORY)
     dku_config.add_param(
         name='gpu_memory_limit',
         value=config.get('gpu_memory_limit') if dku_config.gpu_memory_allocation_mode == 'memory_limit' else 0)
```

Comment:
Still does not work because you compare with a string
```suggestion
        value=config.get('gpu_memory_limit') if dku_config.gpu_memory_allocation_mode == constants.GPU_MEMORY.LIMIT else 0)
```